pkg/config/model.go

```diff
@@ -108,6 +108,9 @@ type Config struct {
 
 	// Status holds long-running status.
 	Status Status
+
+	// Extra holds additional, non-spec information for internal processing.
+	Extra map[string]any
```

Comment:
In krt, we call this "metadata". Should `Meta` hold this information instead of the config?

Reply:
This is really not metadata though.. Meta now holds Kubernetes objects metadata (things like name, namespace, generation, labels, annotations etc.

This is not Kubernetes metadata, rather just a struct for internal processing (which in the future should replace placing a lot of similar hints we are using in internal annotations).

Reply:
It also holds some Istio specific metadata like Domain as well. Not a blocking suggestion though

Reply:
I see. I'd vote to have this separate unless there are concerns with adding `Extra` 

@howardjohn thoughts?

Reply:
I don't care much but Extra lgtm